data/filters/prepare-credit.lua

```diff
@@ -0,0 +1,97 @@
+-- Checks if any contributor information is available
+
+roles = pandoc.List {
+    "conceptualization",
+    "data-curation",
+    "formal-analysis",
+    "funding-acquisition",
+    "investigation",
+    "methodology",
+    "project-administration",
+    "resources",
+    "software",
+    "supervision",
+    "validation",
+    "visualization",
+    "writing-original-draft",
+    "writing-review-editing"
+}
+
+degrees = pandoc.List {
+    "Lead",
+    "Supporting",
+    "Equal"
+}
+
+function invalidRole(str)
+    return not roles:includes(str)
+end
+
+function invalidDegree(str)
+    return not degrees:includes(str)
+end
+
+function join_with_commas_and(list)
+    local len = #list
+    if len == 0 then
+        return ""
+    elseif len == 1 then
+        return list[1]
+    elseif len == 2 then
+        return list[1] .. " and " .. list[2]
+    else
+        local result = table.concat(list, ", ", 1, len - 1)
+        return result .. ", and " .. list[len]
+    end
```

Comment:
add test case for an author with three roles to list them properly? 

also add a test case for an author with no roles when other authors have them

Reply:
I agree that having a proper framework for doing unit tests would be very helpful here to make sure this code does what it's supposed to. Maybe I can add some straight tests into the code that run in the body of this lua file? or is there a better way like we do in Python?

Reply:
i was just thinking of adding some more authors to the sample document, but hey whatever works :)

Reply:
unfortunately, changing the test document creates lots of big diff, so I wanted to avoid this